src/ape_accounts/accounts.py

```diff
@@ -193,20 +209,29 @@ def sign_message(self, msg: Any, **signer_options) -> Optional[MessageSignature]
             for field, value in msg._body_["message"].items():
                 display_msg += f"\t{field}: {value}\n"
 
-            user_approves = self.__autosign or click.confirm(f"{display_msg}\nSign: ")
-
             # Convert EIP712Message to SignableMessage for handling below
             msg = msg.signable_message
+
         elif isinstance(msg, SignableMessage):
-            user_approves = self.__autosign or click.confirm(f"{msg}\n\nSign: ")
+            display_msg = str(msg)
+
         else:
             logger.warning("Unsupported message type, (type=%r, msg=%r)", type(msg), msg)
             return None
 
-        if not user_approves:
+        if sign_raw_hash:
+            # NOTE: Signing a raw hash is so dangerous, we don't want to allow autosigning it
+            if not click.confirm("Please confirm you wish to sign using `EthAccount.signHash`"):
+                return None
+
+            signed_msg = EthAccount.signHash(msg, self.__key)
```

Comment:
**strong recommendation.** this should not happen within this method. There is a reason `EthAccount` kept them separate. the extra prompting, size checking, **leading zeroes ambiguity**

Reply:
Refactored in [fc4b250](https://github.com/ApeWorX/ape/pull/1966/commits/fc4b25035c4672f2e6461a21a7d09fd032b66cbc)